### PR TITLE
Monster/pet carry weight changed to 1/5th of body weight

### PR DESCRIPTION
**Pull request by Davi-DeGanne**
#### Summary
```SUMMARY: Balance "Monster/pet carry weight changed to 1/5th of body weight"```

#### Purpose of change
Previously, carry weight calculation for monsters was the same as for
humans, using a 5 tier size system, where monsters were placed in tiers
based on their weight. This caused a few problems:

- Tier 5 too small in some cases (e.g. cows could only carry 52 kg)
- Tier 1 too big in some cases (e.g. chickens carrying 3x body weight)
- Tiers not granular enough; increase of a single kg could double carry weight

#### Describe the solution
Safe weight capacity of animals seems to be [about 20% - 25% of their body weight](https://en.wikipedia.org/wiki/Pack_animal#Load_carrying_capacity). I went with 20%, for no reason in particular other than it being the more conservative estimate.

#### Describe alternatives you've considered
Making higher load weights possible, but only for limited duration. Seems like a lot of work for a feature most people likely would never use.

### CODE REVIEW

File: src/monster.cpp [hunk 1]
```diff
@@ -2547,6 +2547,11 @@ units::mass monster::get_weight() const
     return units::operator*( type->weight, get_size() / type->size );
 }
 
+units::mass monster::weight_capacity() const
+{
+    return type->weight / 5;
```
Comment (anothersimulacrum) [inline src/monster.cpp]:
```suggestion
    return type->weight * mountable_weight_ratio;
```

**Comment by davidpwbrown** [comment]
This also has the benefit of bringing it in line with the max weight a horse can carry when the player tries to mount them.

I put that at 1/5th the animals weight as the limit there too, so its good itll be consistent now.

As you say - 1/5th is not the perfect solution as they could carry more weight for shorter periods, but itl do for the moment until we have animal stamina and caring / feeding your packmule/mount etc.

**Comment by Fris0uman** [comment]
Could this be somewhat jsonized? So that for exemple animals get the 1/5 ratio by default but you could override it to make stronger monsters?

**Comment by anothersimulacrum** [comment]
That's what the suggestion I made (and was comitted) does.
By default everything will be 20%, but any most er with a mountable weight ration will use that instead.

**Comment by DaviBones** [comment]
Checked the CI failure, seems unrelated to my changes. Let me know if I missed something.

**Comment by yowshi** [comment]
ok i am fairly certain that horses should be able to carry a personm + more then 40kg's at this point i am not at below 35 Kg carry weight and i still can't mount a horse. this is rediculus as horses and cows are now useless as m,ounts.

**Comment by DaviBones** [comment]
You are in the wrong place. This PR buffed horse max inventory weight substantially. And it didn't touch their maximum rider weight at all.

**Pull request merged**